src/core/public/core_app/errors/url_overflow_ui.tsx

```diff
@@ -0,0 +1,101 @@
+import { EuiText } from '@elastic/eui';
+
+import { IBasePath } from '../../http';
+import { IS_IE, URL_MAX_LENGTH } from './url_overflow';
+
+export const UrlOverflowUi: React.FC<{ basePath: IBasePath }> = ({ basePath }) => {
+  return (
+    <EuiText>
+      <p>
+        <FormattedMessage
+          id="common.ui.errorUrlOverflow.errorDescription"
+          defaultMessage="That's a big URL you have there. I have some unfortunate news: Your browser doesn't play nice
+    with Kibana's bacon-double-cheese-burger-with-extra-fries sized URL. To keep you from running
+    into problems Kibana limits URLs in your browser to {urlCharacterLimit} characters for your
+    safety."
+          values={{ urlCharacterLimit: URL_MAX_LENGTH }}
+        />
+      </p>
+
+      <h3>
+        <FormattedMessage
+          id="common.ui.errorUrlOverflow.howTofixErrorTitle"
+          defaultMessage="Ok, how do I fix this?"
+        />
+      </h3>
+      <p>
+        <FormattedMessage
+          id="common.ui.errorUrlOverflow.howTofixErrorDescription"
+          defaultMessage="This usually only happens with big, complex dashboards, so you have some options:"
+        />
+      </p>
+
+      <ol>
+        <li>
+          <FormattedMessage
+            id="common.ui.errorUrlOverflow.howTofixError.enableOptionText"
+            defaultMessage="Enable the {storeInSessionStorageConfig} option in the {kibanaSettingsLink}. This will prevent the URLs from
+      getting long, but makes them a bit less portable."
+            values={{
+              storeInSessionStorageConfig: <code>state:storeInSessionStorage</code>,
+              kibanaSettingsLink: (
+                <a href={basePath.prepend('/app/management/kibana/settings')}>
+                  <FormattedMessage
+                    id="common.ui.errorUrlOverflow.howTofixError.enableOptionText.advancedSettingsLinkText"
+                    defaultMessage="advanced settings"
+                  />
+                </a>
+              ),
+            }}
+          />
+        </li>
+        <li>
+          <FormattedMessage
+            id="common.ui.errorUrlOverflow.howTofixError.removeStuffFromDashboardText"
+            defaultMessage="Remove some stuff from your dashboard. This will reduce the length of the URL and keep your browser in a good place."
+          />
+        </li>
+        {IS_IE && (
+          <li>
+            <FormattedMessage
+              id="common.ui.errorUrlOverflow.howTofixError.doNotUseIEText"
+              defaultMessage="Don't use IE. Every other supported browser we know of doesn't have this limit."
+            />
+          </li>
+        )}
+      </ol>
+      <br />
+      <br />
+      <p>
+        <small>
+          <FormattedMessage
+            id="common.ui.errorUrlOverflow.footnoteText"
+            defaultMessage="Footnote: Party size candy bars are tiny. Party size sub sandwiches are massive. Really makes you think."
+          />
+        </small>
+      </p>
```

Comment:
Is this type of messaging necessary? @gchaps thoughts on things like this?

Reply:
Definitely not 😄 I just left it as-is, but I'm happy to remove this.

Reply:
Removed